gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -681,6 +681,7 @@ private ManagedChannel createSingleChannel() throws IOException {
     builder =
         builder
             .intercept(new GrpcChannelUUIDInterceptor())
+            .intercept(new GrpcLoggingInterceptor())
```

Comment:
just a thought (I don't know if there are any downsides or if this would work): Could we gate adding the logging interceptor here?

i.e. Check if the logging env var exists + is configured and only add an interceptor if so.

I think there was some mention about not using interceptors (potentially) so it may not work in the future.

Reply:
I guess there is no harm in adding a gate here for now.
In the future, another thought is to allow setting enable logging via system properties in addition to env var, in that case, it could change at runtime and this would not work.

Reply:
I was just thinking that having it gate there might be a way to remove all the `if (loggingEnabled) {...}` checks in LoggingUtils.java. But it may not be possible if we know there are future changes that would make this invalid.